python/test/unit/language/test_core.py

```diff
@@ -307,9 +307,12 @@ def kernel(Z, X, Y, SIZE: tl.constexpr):
         y[:] = float('nan')
     # reference result
     z_ref = eval(expr if numpy_expr is None else numpy_expr)
-    dtype_z = _binary_op_dtype_override(dtype_x, dtype_y)
-    if dtype_z is not None:
+    if dtype_z is not None:  # override dtype if specified, for example the comparison operator returns bool
         z_ref = z_ref.astype(dtype_z)
+    else:
+        dtype_z = _binary_op_dtype_override(dtype_x, dtype_y)
+        if dtype_z is not None:
+            z_ref = z_ref.astype(dtype_z)
```

Comment:
interesting, how was the comparison test working before?

Reply:
That's a good point. The root cause behind it is that triton's `int1` has a bitwidth of `1` instead of `8`